jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/toolwindow/SqsWindow.kt

```diff
@@ -3,76 +3,71 @@
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.runBlocking
-import kotlinx.coroutines.withContext
+import com.intellij.ui.content.Content
+import com.intellij.ui.content.ContentManager
 import software.amazon.awssdk.services.sqs.SqsClient
-import software.aws.toolkits.core.utils.error
-import software.aws.toolkits.core.utils.getLogger
 import software.aws.toolkits.jetbrains.core.awsClient
-import software.aws.toolkits.jetbrains.core.toolwindow.ToolkitToolWindowManager
-import software.aws.toolkits.jetbrains.core.toolwindow.ToolkitToolWindowType
 import software.aws.toolkits.jetbrains.services.sqs.Queue
 import software.aws.toolkits.jetbrains.services.sqs.telemetryType
-import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
-import software.aws.toolkits.jetbrains.utils.getCoroutineUiContext
-import software.aws.toolkits.resources.message
 import software.aws.toolkits.telemetry.SqsTelemetry
 
-class SqsWindow(private val project: Project) : CoroutineScope by ApplicationThreadPoolScope("SqsWindow") {
-    private val toolWindow = ToolkitToolWindowManager.getInstance(project, SQS_TOOL_WINDOW)
-    private val edtContext = getCoroutineUiContext()
+class SqsWindow(private val project: Project) {
     private val client: SqsClient = project.awsClient()
 
     fun pollMessage(queue: Queue) {
-        showQueue(queue, SqsWindowUi(project, client, queue).apply { pollMessage() })
+        showQueue(queue).pollMessage()
     }
 
     fun sendMessage(queue: Queue) {
-        showQueue(queue, SqsWindowUi(project, client, queue).apply { sendMessage() })
+        showQueue(queue).sendMessage()
     }
 
-    private fun showQueue(queue: Queue, component: SqsWindowUi) = launch {
+    private fun showQueue(queue: Queue): SqsWindowUi {
         SqsTelemetry.openQueue(project, queue.telemetryType())
-        try {
-            val existingWindow = toolWindow.find(queue.queueUrl)
-            if (existingWindow != null) {
-                withContext(edtContext) {
-                    // We dispose the existing one so we can switch tabs, currently we have no other way to do this
-                    // TODO fix this to switch tabs instead of disposing when we can get the original component out of
-                    // toolwindow
-                    existingWindow.dispose()
-                }
-            }
-
-            withContext(edtContext) {
-                toolWindow.addTab(queue.queueName, component.mainPanel, activate = true, id = queue.queueUrl)
-            }
-        } catch (e: Exception) {
-            LOG.error(e) { "Exception thrown while trying to open queue '${queue.queueName}'" }
-            throw e
+
+        val toolWindow = SqsWindowFactory.getToolWindow(project)
+        val contentManager = toolWindow.contentManager
+
+        val sqsViewContent = findQueue(queue.queueUrl) ?: createSqsView(contentManager, queue)
+
+        toolWindow.show()
+        contentManager.setSelectedContent(sqsViewContent, true)
```

Comment:
does set/show order matter? and does this need to be on edt?

Reply:
It is same order I saw in IJ code, so I don't think so...its either fast enough or it happens in the same paint call?